cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt

```diff
@@ -651,171 +651,13 @@ private constructor(
-         * Collects the requested passes stored in [registeredPasses] and generates a
-         * [PassWithDepsContainer] consisting of pairs of passes and their dependencies.
-         *
-         * @return A populated [PassWithDepsContainer] derived from [registeredPasses].
-         */
-        private fun collectInitialPasses(): PassWithDepsContainer {
-            val workingList = PassWithDepsContainer()
-
-            val softDependencies =
-                mutableMapOf<KClass<out Pass<*>>, MutableSet<KClass<out Pass<*>>>>()
-            val hardDependencies =
-                mutableMapOf<KClass<out Pass<*>>, MutableSet<KClass<out Pass<*>>>>()
-
-            // Add the "execute before" dependencies.
-            for (p in passes) {
-                val executeBefore = mutableListOf<KClass<out Pass<*>>>()
-
-                val depAnn = p.findAnnotations<DependsOn>()
-                // collect all dependencies added by [DependsOn] annotations.
-                for (d in depAnn) {
-                    val deps =
-                        if (d.softDependency) {
-                            softDependencies.computeIfAbsent(p) { mutableSetOf() }
-                        } else {
-                            hardDependencies.computeIfAbsent(p) { mutableSetOf() }
-                        }
-                    deps += d.value
-                }
-
-                val execBeforeAnn = p.findAnnotations<ExecuteBefore>()
-                for (d in execBeforeAnn) {
-                    executeBefore.add(d.other)
-                }
-
-                for (eb in executeBefore) {
-                    passes
-                        .filter { eb == it }
-                        .forEach {
-                            val deps = softDependencies.computeIfAbsent(it) { mutableSetOf() }
-                            deps += p
-                        }
-                }
-            }
-
-            log.info(
-                "The following mermaid graph represents the pass dependencies: \n ${buildMermaid(softDependencies, hardDependencies)}"
-            )
-
-            for (p in passes) {
-                var passFound = false
-                for ((pass) in workingList.getWorkingList()) {
-                    if (pass == p) {
-                        passFound = true
-                        break
-                    }
-                }
-                if (!passFound) {
-                    workingList.addToWorkingList(
-                        PassWithDependencies(
-                            p,
-                            softDependencies[p] ?: mutableSetOf(),
-                            hardDependencies[p] ?: mutableSetOf()
-                        )
-                    )
-                }
-            }
-            return workingList
-        }
-
-        /**
-         * Builds a markdown representation of a pass dependency graph, based on
-         * [Mermaid](https://mermaid.js.org) syntax.
-         */
-        private fun buildMermaid(
```

Comment:
Personally, I found the mermaid output quite useful on multiple occasions. Is there a specific reason to get rid of it?

Reply:
Yes, I don't know how to adapt it to he current code with parallel passes... I'm happy to receive a commit. Probalby at the end of `order()`

Reply:
It's back [bf706dc](https://github.com/Fraunhofer-AISEC/cpg/pull/1645/commits/bf706dc4bde2235751c7d0ce456627955bd11082)